dart/lib/src/scope.dart

```diff
@@ -53,9 +49,10 @@ class Scope {
   SentryUser? get user => _user;
 
   /// Set the current user.
-  set user(SentryUser? user) {
+  FutureOr<void> setUser(SentryUser? user) async {
```

Comment:
We pretty much wanted the clients to either call the methods sync like before, and we would fire and forget the sync down to the native layer, or they could `await` when calling, so the the sync to native is done by then. Maybe we missunderstood `FutureOr<T>`, as there is no case where we are not awaiting internally and returning without a future.

Better to just use `Future<T>` then, as it always can be called without awaiting?

Reply:
FutureOr<T> works best when defining interfaces for implementation by the consumer of it.
That way, the consumer can decide whether he needs the async-ness of a given interface.

Then you can do 
```dart
abstract class Interface {
 FutureOr<void> someMethod();
}

class NonAsyncImpl implements Interface {
  @override
  void someMethod() {
    // do something
  }
}

class AsyncImpl implements Interface {
  @override
  Future<void> someMethod() async {
    // do something
  }
}

List<Interface> interfaces = [
  NonAsyncImpl(),
  AsyncImpl(),
];

for(final interface in interfaces) {
  // NonAsyncImpl is executed directly without awaiting
  final result = interface.someMethod();
  if(result is Future) {
    // AsyncImpl is executed here
    await result;
  }
}
```

> Better to just use Future<T> then, as it always can be called without awaiting?

Yes, for fire and forget, just use `Future`. Even when not awaited, it will get executed eventually.

Reply:
TIL, thank you!

Reply:
You're welcome. It took me some time too 😅 I've noticed there are some places in the SDk where `FutureOr<T>`s are always awaited, but I guess that's not that bad.

Reply:
I've elaborated on that in https://github.com/getsentry/sentry-dart/issues/870